preview/preview.go

```diff
@@ -0,0 +1,131 @@
+package preview
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	logger "github.com/ipfs/go-log"
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/dataset/dsio"
+	"github.com/qri-io/qfs"
+)
+
+var (
+	log = logger.Logger("preview")
+)
+
+const (
+	// MaxNumDatasetRowsInPreview is the highest number of rows a dataset preview
+	// can contain
+	MaxNumDatasetRowsInPreview = 100
+	// MaxReadmePreviewBytes determines the maximum amount of bytes a readme
+	// preview can be. three bytes less than 1000 to make room for an elipsis
+	MaxReadmePreviewBytes = 997
+)
+
+// CreatePreview generates a preview for a dataset version
+// It expects the passed in dataset to have any relevant script files already
+// loaded
+// Preview currently includes:
+//    - body: 100 rows
+//    - readme: first 997 bytes
+//    - meta: all
+//    - commit: all
+//    - structure: all
+//    - stats: none
+//    - viz: all
+//    - transform: all
+func CreatePreview(ctx context.Context, ds *dataset.Dataset) (*dataset.Dataset, error) {
+
+	var err error
+
+	if ds == nil || ds.IsEmpty() {
+		log.Debugf("CreatePreview: empty dataset")
+		return nil, fmt.Errorf("empty dataset")
+	}
+
+	if ds.Readme != nil && ds.Readme.ScriptFile() != nil {
+		ds.Readme.ScriptBytes, err = ioutil.ReadAll(io.LimitReader(ds.Readme.ScriptFile(), MaxReadmePreviewBytes))
+		if err != nil {
+			log.Errorf("Reading Readme: %s", err.Error())
+			return nil, err
+		}
+
+		if len(ds.Readme.ScriptBytes) == MaxReadmePreviewBytes {
+			ds.Readme.ScriptBytes = append(ds.Readme.ScriptBytes, []byte(`...`)...)
+		}
+		ds.Readme.SetScriptFile(nil)
+	}
+
+	if ds.BodyFile() == nil {
+		return nil, fmt.Errorf("no body file")
+	}
```

Comment:
Support not having a body component at all by moving all body file conversion & the final `json.RawMessage` assignment into the inverse of this conditional:

```go
if ds.BodyFile() != nil {
  // ...
```

Add a test to confirm that datasets without a body file can still create previews